### ISSUE TITLE
SyncBatchNorm performance not match torch.nn.BatchNorm2d

### ISSUE THREAD

**Issue by yunlu-chen** [zhanghang1989/PyTorch-Encoding#178]
Hi, 

Thanks for the nice work.

For my own model, I replaced all torch.nn.BatchNorm2d layers with encoding.nn.SyncBatchNorm. But when training on single GPU, it turns out the performance is lower than the bn layer in Pytorch. I also tried to run syncbn on 2 GPUs with the same batch size, the result is the same as syncbn on single GPU, still not comparable to the pytorch bn.

I wonder what can be the difference in behavior in your syncbn with Pytorch bn, or if there can be something wrong with my setup. Any clue on this?

I have double-checked that initialization is the same (weights constant 1 and biases constant 0)
Hyper parameter settings are all default. Pytorch v1.0.1 stable compiled from source.

**Comment by zhanghang1989** [comment]
It is normal. Standard BN use CUDNN, and does not require synchronization. SyncBN use CUDA kernel and synchronization takes some time.

**Comment by yunlu-chen** [comment]
> It is normal. Standard BN use CUDNN, and does not require synchronization. SyncBN use CUDA kernel and synchronization takes some time.

So you mean it is normal that SyncBN and Standard BN outcome different training/evaluation performance as well as time efficiency? 
It was surprising to me that in my case the evaluation error increases quite significantly when switching to SyncBN.

**Comment by zhanghang1989** [comment]
For evaluation, you can simply replace SyncBN with BN.
